### PR TITLE
PCRM-3988 - Add Point-In-Time-Recovery to DynamoDB

### PR DESCRIPTION
**Pull request by abbas-khan8**

### CODE REVIEW

File: terraform/modules/dynamodb/variable.tf [hunk 1]
```diff
@@ -34,3 +34,8 @@ variable "attributes" {
 variable "table_range_key" {
   type = string
 }
+
+variable "recovery_flag" {
```
Comment (joeds13) [inline terraform/modules/dynamodb/variable.tf:38]: I'd name this really obviously to `point_in_time_recovery_enabled` as it links directly to a named piece of AWS DynamoDB functionality, it'll make it obvious to future readers of this code

File: terraform/stacks/list-reconciliation/dynamodb.tf [hunk 2]
```diff
@@ -20,6 +20,7 @@ module "jobs" {
       type = "S"
     }
   ]
+  recovery_flag = true
```
Comment (joeds13) [inline terraform/stacks/list-reconciliation/dynamodb.tf:23]: I'd move this to line 6

I like to group the key + value and the key + object variables in terraform, it makes it easier and consistent to read and often looks better when terraform fmt'd

File: terraform/stacks/list-reconciliation/dynamodb.tf [hunk 3]
```diff
@@ -34,6 +35,7 @@ module "jobs_stats" {
...
```
Comment (joeds13) [inline terraform/stacks/list-reconciliation/dynamodb.tf:38]: I'd move this to line 32

I like to group the key + value and the key + object variables in terraform, it makes it easier and consistent to read and often looks better when terraform fmt'd

**Review by joeds13 — CHANGES_REQUESTED** [review]

**Review by jamesnunnnhs — APPROVED** [review]
Nothing more to add beyond Joe's comments, lgtm

**Comment by josh-scargill-nhsd** [comment]
You've tested this works in AWS right?

**Review by joeds13 — APPROVED** [review]

**Pull request merged**